src/services/codefixes/fixStrictClassInitialization.ts

```diff
@@ -0,0 +1,141 @@
+        return { description, changes, fixId: fixIdAddUndefinedType };
+    }
+
+    function addUndefinedType(changeTracker: textChanges.ChangeTracker, propertyDeclarationSourceFile: SourceFile, propertyDeclaration: PropertyDeclaration): void {
+        const newTypeNode = createUnionTypeNode([propertyDeclaration.type, createKeywordTypeNode(SyntaxKind.UndefinedKeyword)]);
+        changeTracker.replaceNode(propertyDeclarationSourceFile, propertyDeclaration.type, newTypeNode);
+    }
+
+    function getActionsForAddMissingInitializer (context: CodeFixContext, propertyDeclaration: PropertyDeclaration, newLineCharacter: string): CodeFixAction | undefined {
+        const checker = context.program.getTypeChecker();
+        const initializer = getInitializer(checker, propertyDeclaration);
+        if (!initializer) return undefined;
+
+        const description = formatStringFromArgs(getLocaleSpecificMessage(Diagnostics.add_default_Initializer_to_property_0), [propertyDeclaration.name.getText()]);
+        const changes = textChanges.ChangeTracker.with(context, t => addInitializer(t, context.sourceFile, propertyDeclaration, initializer, newLineCharacter));
+        return { description, changes, fixId: fixIdAddInitializer };
+    }
+
+    function addInitializer (changeTracker: textChanges.ChangeTracker, propertyDeclarationSourceFile: SourceFile, propertyDeclaration: PropertyDeclaration, initializer: Expression, newLineCharacter: string): void {
+        const property = updateProperty(
+            propertyDeclaration,
+            propertyDeclaration.decorators,
+            propertyDeclaration.modifiers,
+            propertyDeclaration.name,
+            propertyDeclaration.questionToken,
+            propertyDeclaration.type,
+            initializer
+        );
+        changeTracker.replaceNode(propertyDeclarationSourceFile, propertyDeclaration, property, { suffix: newLineCharacter });
+    }
+
+    function getInitializer(checker: TypeChecker, propertyDeclaration: PropertyDeclaration): Expression | undefined {
+        return getDefaultValueFromType(checker, checker.getTypeFromTypeNode(propertyDeclaration.type));
+    }
+
+    function getDefaultValueFromType (checker: TypeChecker, type: Type): Expression | undefined {
+        if (type.flags & TypeFlags.String) {
+            return createLiteral("");
+        }
+        else if (type.flags & TypeFlags.Number) {
+            return createNumericLiteral("0");
+        }
+        else if (type.flags & TypeFlags.Boolean) {
+            return createFalse();
+        }
+        else if (type.flags & TypeFlags.Literal) {
+            return createLiteral((<LiteralType>type).value);
+        }
+        else if (type.flags & TypeFlags.Union) {
+            return firstDefined((<UnionType>type).types, t => getDefaultValueFromType(checker, t));
+        }
+        else if (type.flags & TypeFlags.Object) {

```

Comment:
Earlier comment marked as out of date, so copying here: `getObjectFlags` will handle any type as input, so no need for `if (type.getFlags() & TypeFlags.Object)` check first.